docs/source/daml/intro/5_Restrictions.rst

```diff
@@ -0,0 +1,183 @@
+
+Failing Actions
+---------------
+
+Not only are ``Update`` and ``Scenario`` examples of ``Action``, they are both examples of actions that can fail, e.g. because a transaction is illegal or the party retrieved via ``getParty`` doesn't exist on the ledger.
+
+Each has a special action ``abort txt`` which takes on type ``Update ()`` or ``Scenario ()`` depending on context and represents failure. Transactions and scenarios succeed or fail *atomically* as a whole so an occurrence of an ``abort`` action will always fail the entire evaluation of the current ``Scenario`` or ``Update``.
+
+The last expression in the ``do`` block of the ``Redeem`` choice is a pattern matching expression on ``dow``. It has type ``Update ()`` and is either an ``abort`` or ``return`` depending on the day of week. So during the week, it's a no-op and on weekends, it's the special failure action. Thanks to the atomicity of transactions, no transaction can ever make use of the ``Redeem`` choice on weekends.
+
+A sample Action
+---------------
+
+If the above didn't make complete sense, here's another example. ``CoinGame a`` is an ``Action a`` in which a ``Coin`` is flipped and based on the ``Heads`` and ``Tails`` results, a value of type ``a`` is calulated.
+
+.. literalinclude:: daml/daml-intro-5/Restrictions.daml
+  :language: daml
+  :start-after: -- COIN_BEGIN
+  :end-before: -- COIN_END
+
+A ``CoinGame a`` exposes a function ``play`` which takes a ``Coin`` and returns a new ``Coin`` and a result ``a``. More on the ``->`` syntax for functions later.
+
+``Coin`` and ``play`` are deliberately left obscure in the above. All you have is an action ``getCoin`` to get your hands on a ``Coin`` in a ``Scenario`` context and an action ``flipCoin`` which represents the simplest possible game: A single coin flip resulting in a  ``Face``. You can't play any ``CoinGame`` game on pen and paper as you don't have a coin, but you can write down a script or recipe for a game:
+
+.. literalinclude:: daml/daml-intro-5/Restrictions.daml
+  :language: daml
+  :start-after: -- COIN_SCENARIO_BEGIN
+  :end-before: -- COIN_SCENARIO_END
+
+The ``game`` expression is a ``CoinGame`` in which a coin is flipped three times. If all three tosses return ``Heads``, the result is ``"Win"``, or else ``"Loss"``.
+
+In a ``Scenario`` context you can get a ``Coin``, which is actually a pseudo-random number generator based on LET, and play the game.
+
+*Somehow* the ``Coin`` is threaded through the various actions. If you want to look through the looking glass and understand in-depth what's going on, you can look at the source file to see how the ``CoinGame`` action is implemented, though be warned that the implementation uses a lot of DAML features we haven't introduced yet in this introduction.
+
+More generally, if you want to learn more about Actions (aka Monads), we recommend a general course on functional programming, and Haskell in particular.
```

Comment:
This isn't super useful without linking to a specific resource, I'd say - can we recommend eg Learn you a Haskell, even linking to the relevant chapter?

Reply:
Will try to crowdsource some suggestions. I like Learn you a Haskell, but apparently it's a love it/hate it sort of book.